### PR TITLE
AST-2008 - Customizer: Woocommerce Cart Click Action "Cart Page"

### PR DESCRIPTION
**Pull request by neerajmasai**
### Description

-  New option in "Cart Click Action" -> "Cart Page"
- Upon activation, will redirect user to Cart Page
- Shortened other option names "Dropdown Cart" -> "Dropdown" and "Slide-In Cart" -> "Slide In" to accomodate space for new option.

### Screenshots

https://d.pr/v/XFA3AT

### Types of changes

- modified option, cart click action with new choice: 'Cart Page'.
- changes in markup, to redirect to cart page.

### How has this been tested?

-  In customizer > header builder > cart > cart click action
- check to see if a new option 'Cart Page' exists
- Click on it to activate
- now refresh website or in customizer preview, click on cart icon
- It should redirect to shopping cart page
- also tested with other options, "Dropdown", "Slide In", to check if changes do not break anything.

### Checklist:
- [x] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: inc/customizer/configurations/builder/header/class-astra-customizer-woo-cart-configs.php [hunk 1]
```diff
@@ -139,8 +139,9 @@ public function register_configuration( $configurations, $wp_customize ) {
 				'control'    => 'ast-selector',
 				'priority'   => 60,
 				'choices'    => array(
-					'default' => __( 'Dropdown Cart', 'astra' ),
-					'flyout'  => __( 'Slide-In Cart', 'astra' ),
+					'default' => __( 'Dropdown', 'astra' ),
+					'flyout'  => __( 'Slide-In', 'astra' ),
+					'redirect_to_cart' => __( 'Cart Page', 'astra' ),
```
Comment (nathanrodrigues2111) [inline inc/customizer/configurations/builder/header/class-astra-customizer-woo-cart-configs.php:144]: Try to align all the arrows "=>" in one line for better readability

Comment (neerajmasai) [reply inc/customizer/configurations/builder/header/class-astra-customizer-woo-cart-configs.php:144]: Alright, Something like this?
					'flyout'  => __( 'Slide-In Cart', 'astra' ),
					'default' => __( 'Dropdown', 'astra' ),
					'flyout'  => __( 'Slide-In', 'astra' ),
					'redirect' => __( 'Cart Page', 'astra' ),

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 2]
```diff
@@ -2184,20 +2184,33 @@ public function woo_mini_cart_markup() {
...
 
 			ob_start();
 			if ( is_customize_preview() && true === Astra_Builder_Helper::$is_header_footer_builder_active ) {
 				Astra_Builder_UI_Controller::render_customizer_edit_button();
 			}
 			?>
-			<div id="ast-site-header-cart" class="ast-site-header-cart <?php echo esc_attr( implode( ' ', $cart_menu_classes ) ); ?>">
-				<div class="ast-site-header-cart-li <?php echo esc_attr( $class ); ?>">
-					<?php $this->astra_get_cart_link(); ?>
+			<?php if($redirect_cart == "redirect_to_cart"): ?>
```
Comment (nathanrodrigues2111) [inline inc/compatibility/woocommerce/class-astra-woocommerce.php]: Please use Yoda condition : https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#yoda-conditions

File: inc/customizer/configurations/builder/header/class-astra-customizer-woo-cart-configs.php (hunk 1 above)
Comment (nathanrodrigues2111) [inline inc/customizer/configurations/builder/header/class-astra-customizer-woo-cart-configs.php:144]: Try to align all the arrows "=>" in one line for better readability

**Review by nathanrodrigues2111 — APPROVED** [review]

File: inc/lib/astra-notices/class-astra-notices.php [hunk 3]
```diff
@@ -116,7 +116,7 @@ public function dismiss_notice() {
 			}
 
 			if ( false === wp_verify_nonce( $nonce, 'astra-notices' ) ) {
-				wp_send_json_error( esc_html_e( 'WordPress Nonce not validated.', 'astra' ) );
+				wp_send_json_error( esc_html_e( 'WordPress Nonce not validated.' ) );
```
Comment (imnavanath) [inline inc/lib/astra-notices/class-astra-notices.php:119]:
```suggestion
				wp_send_json_error( esc_html_e( 'WordPress Nonce not validated.', 'astra' ) );
```

**Comment by neerajmasai** [comment]
PR comments have been resolved. Please review. @imnavanath

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 4]
```diff
@@ -2226,9 +2234,9 @@ public function header_section_elements( $options ) {
 		 */
 		public function astra_get_cart_link() {
 			$view_shopping_cart = apply_filters( 'astra_woo_view_shopping_cart_title', __( 'View your shopping cart', 'astra' ) );
-
+			$cart_page           = astra_get_option( 'woo-header-cart-click-action' );
 			$woo_cart_link = wc_get_cart_url(); 
-			if ( is_customize_preview() ) {
+			if ( is_customize_preview() && "cart_page" !== $cart_page) {
```
Comment (imnavanath) [inline inc/compatibility/woocommerce/class-astra-woocommerce.php:2239]:
```suggestion
			if ( is_customize_preview() && 'cart_page' !== $cart_click_action ) {
```

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 5]
```diff
@@ -2226,9 +2234,9 @@ public function header_section_elements( $options ) {
 		 */
 		public function astra_get_cart_link() {
 			$view_shopping_cart = apply_filters( 'astra_woo_view_shopping_cart_title', __( 'View your shopping cart', 'astra' ) );
-
+			$cart_page           = astra_get_option( 'woo-header-cart-click-action' );
```
Comment (imnavanath) [inline inc/compatibility/woocommerce/class-astra-woocommerce.php:2237]:
```suggestion
			$cart_click_action           = astra_get_option( 'woo-header-cart-click-action' );
```

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 6]
```diff
@@ -2195,7 +2196,14 @@ public function woo_mini_cart_markup() {
 					<?php $this->astra_get_cart_link(); ?>
 				</div>
 				<div class="ast-site-header-cart-data">
-					<?php the_widget( 'WC_Widget_Cart', 'title=' ); ?>
+					<?php 
+						// if cart page is active then do not display widget.
+					if ( 'cart_page' !== $cart_page ) {
```
Comment (imnavanath) [inline inc/compatibility/woocommerce/class-astra-woocommerce.php:2201]:
```suggestion
					if ( 'cart_page' !== $cart_click_action ) {
```

**Comment by imnavanath** [comment]
@neerajmasai code analysis is failed, please take a look.

**Comment by neerajmasai** [comment]
Newer PR solved this issue.

**Pull request closed without merging**